pkg/plugins/backendplugin/manager/plugin_settings.go

```diff
@@ -20,9 +20,12 @@ func (ps pluginSettings) ToEnv(prefix string, hostEnv []string) []string {
 	return env
 }
 
-func extractPluginSettings(cfg *setting.Cfg) map[string]pluginSettings {
-	psMap := map[string]pluginSettings{}
-	for pluginID, settings := range cfg.PluginSettings {
+func getPluginSettings(plugID string, cfg *setting.Cfg) pluginSettings {
+	for id, settings := range cfg.PluginSettings {
+		if id != plugID {
+			continue
+		}
```

Comment:
I think this should be possible to do in `O(map lookup)` instead of `O(n)`, right?

```suggestion
	settings, exists := cfg.PluginSettings[plugID]
	if !exists {
		return nil
	}
```

Reply:
@sakjur let me look again.

Reply:
@sakjur please have a look at my revision which is even simpler.